### PR TITLE
Tips in general coding + merge chapters 4 and 5 into a base r chapter

### LINKED ISSUE #14: Chapter 3 adding good coding practice
I think it'd be good to do a section on what we see as good coding practice. @adamrobinson361  mentioned yesterday having a good **readme** in each project to give users an idea of what's going on. I agree that's a good idea, but also think we should have some set style of teaching coding. @matt-dray  mentioned a shortcut to add -------'s across code to break sections. Would be good to hear both your thoughts on this one!

### LINKED ISSUE #22: Convert Chapter 4 to Base R
Convert chapter 4 such that it is called Base R and covers at a high level the base R functionality.

### PR DESCRIPTION
**Pull request by adamrobinson361**
To resolve #14 and to resolve #22

### CODE REVIEW

File: 04-base-r.Rmd [hunk 1]
```diff
@@ -1,4 +1,109 @@
...
+
+This chapter summarises some of the core functions for manipulating data in base R.
+
+## Loading in data
+
+The most common process to load in data is to load in a CSV file.
+
+```{r sec4_1,echo=FALSE}
+#Load in data
+swfc_16_init <- read.csv("data/SWFC_2016_Machine_Readable.csv")
```
Comment (TomFranklin) [inline 04-base-r.Rmd]: At this point of loading data, can we add to reading in .csv's - `stringsasfactors = FALSE` ? Will help a lot with the consistency of the dplyr chapter 6!

Comment (TomFranklin) [reply 04-base-r.Rmd]: And same for the 2nd loading .csv - will just really help when I'm describing variable types

Comment (adamrobinson361) [reply 04-base-r.Rmd]: I don’t think we should do that - factors are useful for stuff like regression. Can you make clear in the next chapter that when you are not running statistical models that factors tend to be of less use instead

Comment (TomFranklin) [reply 04-base-r.Rmd]: Yeah that's fine. It was mainly for summary statistics and then casting variables etc.

**Pull request merged**